scipy/optimize/_numdiff.py

```diff
@@ -344,34 +365,81 @@ def fun_wrapped(x):
     if np.any((x0 < lb) | (x0 > ub)):
         raise ValueError("`x0` violates bound constraints.")
 
-    h = _compute_absolute_step(rel_step, x0, method)
+    if as_linear_operator:
+        if rel_step is None:
+            rel_step = _compute_relative_step(method)
+
+        return _linear_operator_difference(fun_wrapped, x0,
```

Comment:
Thank you a lot for this comment @nmayorov! It gave me insight on things I was not considering before, yielding significant improvements on the implementation.